hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/expression/json/JsonQueryFunctionIntegrationTest.java

```diff
@@ -103,44 +103,46 @@ public void test_extendedSyntax() {
     public void test_defaultWrapperBehavior() {
         initComplexObject();
 
-        assertEquals("[1,\"2\",3,{\"t\":1}]",
-                querySingleValue("SELECT JSON_QUERY(this, '$[0]') FROM test").toString());
-        assertEquals("{\"t\":1}",
-                querySingleValue("SELECT JSON_QUERY(this, '$[1]') FROM test").toString());
-        assertThatThrownBy(() -> querySingleValue("SELECT JSON_QUERY(this, '$[2]' ERROR ON ERROR) FROM test"))
-                .isInstanceOf(HazelcastSqlException.class)
-                .hasMessageContaining("JSON_QUERY result is not an array or object");
+        assertEquals(json("[1,\"2\",3,{\"t\":1}]"),
+                querySingleValue("SELECT JSON_QUERY(this, '$[0]') FROM test"));
+        assertEquals(json("{\"t\":1}"),
+                querySingleValue("SELECT JSON_QUERY(this, '$[1]') FROM test"));
+        assertEquals(json("3"),
```

Comment:
Think I missed this, but `JSON_QUERY` is not supposed to ever return a primitive (non-Object/Array) value. 
With the default `WRAPPER` value (`WITH CONDITIONAL WRAPPER` iirc) its supposed to return `[3]` here.

Reply:
Hmm, I double checked the spec, and indeed, it's not supposed to return a JSON literal. However, Oracle supports it and IMO, it makes perfect sense. A standalone JSON literal is a well-formed JSON document:
![image](https://user-images.githubusercontent.com/9109877/148185824-665c211b-fb9e-4897-a466-d2f20f8a71fb.png)

`JSON_QUERY` returns a json documents. IMO the restriction that the JSON document can be only json-null, array and object is arbitrary. Why exclude literals?

The difference between JSON_QUERY and JSON_VALUE on literals is also illustrated here:
![image](https://user-images.githubusercontent.com/9109877/148184818-c530c61a-298c-4c9b-a3a0-ed8063e7dc21.png)

I'm for keeping it. It's an extension of the standard, but one that IMO should have been there from the beginning. Maybe it's even in some newer release...